plugins/org.eclipse.elk.alg.packing.rectangles/src/org/eclipse/elk/alg/packing/rectangles/firstiteration/FirstIteration.java

```diff
@@ -0,0 +1,341 @@
+        candidates.add(opt1);
+        candidates.add(opt2);
+        candidates.add(opt3);
+        candidates.add(opt4);
+
+        switch (packingStrat) {
+        // examines the candidates according to the packing strategy.
+        case MAX_SCALE_DRIVEN:
+            candidates = BestOptionFinder.findMaxScale(candidates);
+            if (candidates.size() > 1) {
+                candidates = BestOptionFinder.findMinArea(candidates);
+                if (candidates.size() > 1) {
+                    candidates = BestOptionFinder.findBestAspectRatio(candidates, dar);
+                }
+            }
+            break;
+        case ASPECT_RATIO_DRIVEN:
+            candidates = BestOptionFinder.findBestAspectRatio(candidates, dar);
+            if (candidates.size() > 1) {
+                candidates = BestOptionFinder.findMinArea(candidates);
+                if (candidates.size() > 1) {
+                    candidates = BestOptionFinder.findMaxScale(candidates);
+                }
+            }
+
+            break;
+        case AREA_DRIVEN:
+            candidates = BestOptionFinder.findMinArea(candidates);
+            if (candidates.size() > 1) {
+                candidates = BestOptionFinder.findMaxScale(candidates);
+                if (candidates.size() > 1) {
+                    candidates = BestOptionFinder.findBestAspectRatio(candidates, dar);
+                }
+            }
+
+            break;
+        default:
+            break;
+        }
+        // only one candidate remains
+        if (candidates.size() == 1) {
+            return candidates.get(candidates.size() - 1);
+        }
+
+        // Multiple options have the same value for every benchmark. These special cases are caught in the following.
+        if (candidates.size() == 2) {
+            return checkSpecialCases(candidates.get(0), candidates.get(1), lastPlaced, toPlace);
+        }
+
+        return null;
+    }
+
+    /**
+     * Checks the two options and their possible combinations after being sorted out be the packing strategies.
+     * Determines the best option depending on the options given.
+     * 
+     * @param drawing1
+     *            a drawing to compare.
+     * @param drawing2
+     *            a drawing to compare.
+     * @param lastPlaced
+     *            the rectangle that was last placed before the two drawings were calculated.
+     * @param toPlace
+     *            the rectangle that was newly placed in the two drawings.
+     * @return the better drawing out of the two.
+     */
+    private DrawingData checkSpecialCases(final DrawingData drawing1, final DrawingData drawing2,
+            final ElkNode lastPlaced, final ElkNode toPlace) {
+        DrawingDataDescriptor firstOpt = drawing1.getPlaOpt();
+        DrawingDataDescriptor secondOpt = drawing2.getPlaOpt();
+
+        DrawingDataDescriptor lpRight = DrawingDataDescriptor.CANDIDATE_POSITION_LAST_PLACED_RIGHT;
```

Comment:
Again, don't use variables for constants. It confuses the reader as to whether the variables are actually constant or have been computed.